sdcm/provision/aws/provisioner.py

```diff
@@ -0,0 +1,290 @@
+            instance_parameters: AWSInstanceParams,
+            count: int,
+            tags: Union[List[TagsType], TagsType] = None,
+            names: List[str] = None) -> List[Instance]:
+        if tags is None:
+            tags = {}
+        if isinstance(tags, dict):
+            tags = [tags] * count
+        elif isinstance(tags, list):
+            tags = tags.copy()
+
+        if names:
+            assert len(names) == count, "Names length should be equal to count"
+        assert len(tags) == count, "Tags length should be equal to count"
+
+        for node_id, name in enumerate(names):
+            tag = tags[node_id]
+            tag['Name'] = name
+
+        if isinstance(provision_parameters, ProvisionGuaranteedParams):
+            if provision_parameters.price:
+                raise ValueError("AWS ProvisionGuaranteed(on-demand) does not support price targeting")
+            if provision_parameters.duration:
+                LOGGER.debug("AWS ProvisionGuaranteed(on-demand) does not support duration")
+            return self._provision_guaranteed(
+                provision_parameters=provision_parameters,
+                instance_parameters=instance_parameters,
+                count=count,
+                tags=tags,
+            )
+        if isinstance(provision_parameters, ProvisionNotGuaranteedParams):
+            return self._provision_not_guaranteed(
+                provision_parameters=provision_parameters,
+                instance_parameters=instance_parameters,
+                count=count,
+                tags=tags,
+            )
+        raise RuntimeError("Unexpected provision_parameters type")
+
+    @staticmethod
+    def _is_provision_type_fleet(count: int) -> bool:
+        return count > SPOT_CNT_LIMIT
+
+    def _provision_instance_limit(self, count: int) -> int:
+        return SPOT_FLEET_LIMIT if self._is_provision_type_fleet(count) else SPOT_CNT_LIMIT
+
+    @property
+    def _spot_valid_until(self) -> datetime.datetime:
+        return datetime.datetime.now() + datetime.timedelta(minutes=SPOT_REQUEST_TIMEOUT / 60 + 5)
+
+    @staticmethod
+    def _ec2_client(provision_parameters: ProvisionParamsBase) -> EC2Client:
+        return ec2_clients[provision_parameters.region_name]
+
+    @staticmethod
+    def _full_availability_zone_name(provision_parameters: ProvisionParamsBase) -> str:
+        return provision_parameters.region_name + provision_parameters.availability_zone
+
+    @staticmethod
+    def _spot_block_duration(provision_parameters: ProvisionParamsBase) -> Optional[int]:
+        """
+        A period in minutes AWS will hold spot instance for us.
+        AWS have limit on it, if test is going to be run
+        """
+        if provision_parameters.duration is None:
+            return None
+        if provision_parameters.duration >= MAX_SPOT_DURATION_TIME:
+            LOGGER.debug("Requested stop time more than AWS limit (360 minutes). Spot duration is going to be ignored")
```

Comment:
Better to replace 360 with MAX_SPOT_DURATION_TIME

Reply:
It is referenced from two places, both them need this particular number

Reply:
But if this value changes one day, the log message will need to be change manually.

Reply:
Aaah, i finally got what you mean. fixed